projects/js-packages/components/components/icons/index.tsx

```diff
@@ -241,28 +244,43 @@ const iconsMap = {
 	star: StarIcon,
 	videopress: VideopressIcon,
 	jetpack: JetpackIcon,
+};
+
+const socialIcons = {
 	facebook: FacebookIcon,
 	twitter: TwitterIcon,
 	linkedin: LinkedinIcon,
 	tumblr: TumblrIcon,
 	google: GoogleIcon,
 };
 
+const iconsMap = {
+	...jetpackIcons,
+	...socialIcons,
+};
+
+export type IconsMap = typeof iconsMap;
+
+export type IconSlug = keyof IconsMap;
+
 /**
  * Return icon component by slug.
  *
  * @param {string} slug       - Icon slug.
- * @returns {React.Component}   Icon component.
+ * @returns {React.ComponentType<BaseIconProps>}   Icon component.
  */
-export function getIconBySlug( slug ) {
+export function getIconBySlug< Slug extends IconSlug >( slug: Slug ): IconsMap[ Slug ] {
```

Comment:
`getIconBySlug.displayName` returns `getIconBySlug`, adding an icon to the _stories_ that just shows "getBySlug" in the screenshot.

![Screenshot_2022-07-26_15-15-24](https://user-images.githubusercontent.com/8486249/181081353-81610ddb-2299-4ba4-ae80-52a5036a32af.png)

Reply:
Previously it returned `undefined` and I'm not sure why this changed. Any ideas?

Reply:
Good catch @dhasilva! I have no idea why storybook does that. Maybe it parses JS and TS files differently. Anyways, I have applied a fix to not render that as an icon.